core/src/main/scala/kafka/server/metadata/BrokerMetadataListener.scala

```diff
@@ -248,6 +248,24 @@ class BrokerMetadataListener(
     }
   }
 
+  // This is used in tests to alter the publisher that is in use by the broker.
+  def alterPublisher(publisher: MetadataPublisher): CompletableFuture[Void] = {
+    val event = new AlterPublisherEvent(publisher)
+    eventQueue.append(event)
+    event.future
+  }
+
+  class AlterPublisherEvent(publisher: MetadataPublisher)
+    extends EventQueue.FailureLoggingEvent(log) {
+    val future = new CompletableFuture[Void]()
+
+    override def run(): Unit = {
+      _publisher = Some(publisher)
+      log.info(s"Set publisher to ${publisher}")
```

Comment:
Why info? Seems more like a debug level message.

Reply:
This is only used in unit tests. I think it's good to have it in the log4j since earlier events may have been published differently by the earlier publisher.

Reply:
If this is only used in tests does it make sense to alter `alterPublisher` to give it a second parameter accepting a value that implements `EventQueue.FailureLoggingEvent` and then define the `AlterPublisherEvent` class in `/core/src/test` rather than `core/src/main`.

That way its more clear the class is only being used for tests.